text/0000-or-patterns.md

```diff
@@ -746,4 +833,16 @@ There is support for or-patterns in [various lisp libraries][lisp_libs].
 # Unresolved questions
 [unresolved]: #unresolved-questions
 
-There are none.
+1. Should we allow `top_pat` or `pat<allow_top_alt>` in `inferrable_param` such
+   that closures permit `|Ok(x) | Err(x)|` without first wrapping in parenthesis?
+   We defer this decision to stabilization as it may depend on experimentation.
```

Comment:
I very much think we should require parentheses here. Even if rustc can parse this, as a *human* I would have a hard time with `|` as delimiter wrapped around `|` as alternative.

Reply:
I don't actually think `rustc` could parse this - it's looking for a `|` to close the list of closure parameters, so it would need backtracking to actually tell between patterns and expressions, and that's probably not even enough, since you can write e.g. `|Ok(x)| Err(x)||y|z`.